### PR TITLE
Snap 2474 | Prune partition of row table

### PR DESCRIPTION
**Pull request by PradeepSurale**
## Changes proposed in this pull request

- Added a support for the partition pruning for the Row Table
 
## Patch testing

- Added a scala test which ensures functionality in embedded mode.
- Added Dunit test which ensures functionality in SmartConnector Mode.
- Testing the Pre-checkin

## ReleaseNotes.txt changes

No.

## Other PRs 

No

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/sql/execution/columnar/impl/JDBCSourceAsColumnarStore.scala [hunk 1]
```diff
@@ -1001,10 +1003,22 @@ class SmartConnectorRowRDD(_session: SnappySession,
...
+    if(parts.length == _partEval.apply().length){
+      return getPartitionEvaluator
+    }
     logDebug(s"$toString.getPartitions: $tableName partitions ${parts.mkString("; ")}")
     parts
   }
 
+  def getPartitionEvaluator: Array[Partition] = {
+    partitionPruner() match {
+      case -1 => _partEval.apply()
```
Comment (dshirish) [inline core/src/main/scala/org/apache/spark/sql/execution/columnar/impl/JDBCSourceAsColumnarStore.scala]: _partEval()

File: cluster/src/dunit/scala/org/apache/spark/sql/SmartConnectorFunctions.scala [hunk 2]
```diff
@@ -90,4 +94,109 @@ object SmartConnectorFunctions {
...
+
+  val query6 = "select O_ORDERDATE, {fn TIMESTAMPADD(SQL_TSI_DAY," +
+    " {fn FLOOR((-1 * {fn DAYOFYEAR(O_ORDERDATE)} - 1))}, O_ORDERDATE)}" +
+    " from orders where O_ORDERKEY = 32"
+
+  private def verifyRowTablePruning(snc: SnappyContext): Unit = {
+
+    val earlierValue = io.snappydata.Property.ColumnBatchSize.get(snc.sessionState.conf)
+    try {
+      io.snappydata.Property.ColumnBatchSize.set(snc.sessionState.conf, "1000")
```
[the remark this answers is not in the document]
Comment (PradeepSurale) [reply cluster/src/dunit/scala/org/apache/spark/sql/SmartConnectorFunctions.scala]: Removed the batch size as not needed.

File: cluster/src/test/scala/org/apache/spark/sql/SingleNodeTest.scala [hunk 3]
```diff
@@ -111,83 +146,102 @@ object SingleNodeTest {
...
+    assert(r.getDate(0).toString.equals("1995-07-16"))
+    assert(r.getDate(1).toString.equals("1994-12-30"))
+  }
 
-    df = snc.sql("select * from orders where o_orderkey = {fn substring('d1xxd2', 2, 1)} ")
-    assert(df.collect()(0).getLong(0) == 1)
+  /**
+   *
+   */
+  private def testRowTablePruning(snc: SnappyContext): Unit = {
```
[the remark this answers is not in the document]
Comment (PradeepSurale) [reply cluster/src/test/scala/org/apache/spark/sql/SingleNodeTest.scala]: Corrected.. Removed the duplicate code.

File: core/src/main/scala/org/apache/spark/sql/collection/Utils.scala [hunk 4]
```diff
@@ -90,7 +93,7 @@ object Utils {
   }
 
   def analysisException(msg: String,
-      cause: Option[Throwable] = None): AnalysisException =
+                        cause: Option[Throwable] = None): AnalysisException =
```
Comment (dshirish) [inline core/src/main/scala/org/apache/spark/sql/collection/Utils.scala]: Could you remove the differences in space? Many places in this file.

**Comment by dshirish** [comment]
Added a comment about unnecessary changes space

**Pull request merged**